kore/app/share/GlobalMain.hs

```diff
@@ -560,3 +565,17 @@ mainPatternParseAndVerify indexedModule patternFileName =
 -}
 mainPatternParse :: String -> Main ParsedPattern
 mainPatternParse = mainParse parseKorePattern
+
+{- | Extract axioms from a module and add them to the main definition module.
+ This should be safe, as long as the axioms only depend on sorts/symbols
+ defined in the main module.
+-}
+addExtraAxioms ::
+    VerifiedModule StepperAttributes ->
+    VerifiedModule StepperAttributes ->
+    VerifiedModule StepperAttributes
+addExtraAxioms definitionModule moduleWithExtraAxioms =
+    Lens.over
+        (field @"indexedModuleAxioms")
+        (<> indexedModuleAxioms moduleWithExtraAxioms)
+        definitionModule
```

Comment:
Alternatives:

Without generic-lens:
```haskell
let definitionAxioms = indexedModuleAxioms definitionModule
    extraAxioms = indexedModuleAxiom moduleWithExtraAxioms
 in definitionModule{ indexedModuleAxioms = definitionAxioms <> extraAxioms }
```

With generic-lens:
Lens provides an operator [`<>~`](https://hackage.haskell.org/package/lens-5.1/docs/Control-Lens-Operators.html#v:-60--62--126-) to target semigroups: 
```haskell
definitionModule & field @"indexedModuleAxioms <>~ indexedModuleAxioms moduleWithExtraAxioms
```

All three are fine by me, I'm just being thorough. I like the last option because it is specific to semigroups. `let .. in` is a bit wordy but it avoids the extra generic-lens dependency.

Reply:
I added your second suggestion.